### DISCUSSION TITLE
No response after random period of time

### DISCUSSION THREAD

**Discussion by timbueno** [HomeSpan/HomeSpan#952] [Q&A]
Hello! I've built a very simple HomeSpan accessory that enables and disables an LED via HomeKit and the integrated button on the M5Nano-C6 device using the latest versions of all libraries. 

The code appears sound and works well for random periods of times. Yesterday I was able to interact with the light via HomeKit for ~6 hours until I discovered it was in the "No Response" status. I restarted it by pulling the power and it came back up like normal. It lasted all the way through the night for a total of about 16 hours until it randomly went into the "No Response" state as well.

The integrated button still allows me to turn on and off the LED during the "No Response" state and the device is still connected to the WiFi.

I've attached my code for reference purposes here. 

Have there been any other reports of instability with the new 2.0.0 version of the library and/or C6 devices?

[DEV_LED.h.txt](https://github.com/user-attachments/files/17850423/DEV_LED.h.txt)
[StratLight.ino.txt](https://github.com/user-attachments/files/17850424/StratLight.ino.txt)

**Comment by delingren** [comment]
When you said the device was still connected to the WiFi, how did you verify that? Did you do an "s" over serial to check the status? You can use a USB to TTL adapter to connect to the serial port of a running ESP32. You can also use the USB port itself if the ESP32 is powered directly via the pins. You probably want to use a USB cable with the Vcc wire snipped though, to prevent potential damage to your computer.

**Reply by timbueno** [reply]
My issue seems to have resolved itself. I was using version 3.0.7 of the Arduino-ESP32 Board Manager. I rolled back to 3.0.5 which is the latest version that’s been tested according to the ReadMe and since rolling it back (over 10 days now), it hasn’t displayed the dreaded “No Response” error.

There might be something up with C6 devices and the Arduino-ESP32 Board Manager for versions >3.0.5 but I’m not versed enough to look into it unfortunately.

I’ll keep an eye out and report back if I get the issue again on the 3.0.5 board manager.
